platform/ios/src/MGLMapView.mm

```diff
@@ -3003,10 +3069,10 @@ - (void)locationManager:(__unused CLLocationManager *)manager didUpdateLocations
 
     if (! oldLocation || ! CLLocationCoordinate2DIsValid(oldLocation.coordinate) || [newLocation distanceFromLocation:oldLocation])
     {
-        self.userLocation.location = newLocation;
-
-        // deselect user if applicable since we don't do callout tracking yet
-        if ([self.selectedAnnotation isEqual:self.userLocation]) [self deselectAnnotation:self.userLocation animated:NO];
+        if (self.userTrackingState != MGLUserTrackingStateBegan)
+        {
+            self.userLocation.location = newLocation;
```

Comment:
Only for the transition to the initial location when user tracking is switched on. That transition may take longer than a second and shouldn't be interrupted.